hera_cal/reflections.py

```diff
@@ -267,25 +239,445 @@ def write_reflections(self, output_calfits, input_calfits=None, overwrite=False,
+                Holds time_array of input u modes.
+            full_times : 1D array
+                time_array to interpolate onto. Default is times.
+            uflags : DataContainer
+                Object to pull target u flags from. Default is None.
+            keys : list of tuples
+                List of baseline-pol DataContainer tuples to operate on.
+            overwrite : bool
+                If True, overwrite output data if it exists.
+            verbose : bool
+                If True, report feedback to stdout.
+            mode : str
+                Interpolation mode. Options=['gpr']
+            gp_len : length-scale of GPR in units of times
+            gp_nl : GPR noise-level in units of input u.
+        """
+        if not hasattr(self, 'umode_interp'):
+            self.umode_interp = DataContainer({})
+
+        if uflags is None:
+            uflags = self.uflags
+
+        if full_times is None:
+            full_times = times
+
+        # get keys
+        if keys is None:
+            keys = u.keys()
+
+        # setup X predict
+        Xmean = np.median(times)
+        Xpredict = full_times[:, None] - Xmean
+
+        # iterate over keys
+        for k in keys:
+            # check overwrite
+            if k in self.umode_interp and not overwrite:
+                echo("{} in umode_interp and overwrite == False, skipping...".format(k), verbose=verbose)
+                continue
+
+            if mode == 'gpr':
+                # setup GP kernel
+                kernel = 1**2 * gp.kernels.RBF(length_scale=gp_len) + gp.kernels.WhiteKernel(noise_level=gp_nl)
+                GP = gp.GaussianProcessRegressor(kernel=kernel, optimizer=optimizer, normalize_y=True)
+
+                # setup regression data: get unflagged data
+                select = ~np.max(uflags[k], axis=1)
+                X = times[select, None] - Xmean
+                y = u[k][select, :]
+
+                # fit gp and predict
+                GP.fit(X, y.real)
+                ypredict_real = GP.predict(Xpredict)
+                GP.fit(X, y.imag)
+                ypredict_imag = GP.predict(Xpredict)
+
+                # append
+                self.umode_interp[k] = ypredict_real.astype(np.complex) + 1j * ypredict_imag
+
+            else:
+                raise ValueError("didn't recognize interp mode {}".format(mode))
+
+    def project_autos_onto_u(self, keys, auto_keys, u=None, flags=None, index=0, auto_delay=0,
+                             overwrite=False, verbose=True):
+        """
+        Project autocorr onto u modes.
+
+        Projects the time dependent dfft of an autocorrelation
+        at a specified delay onto the specified u-mode, replacing
+        the u-mode with the projected autocorrelation. Inserts
+        results into self.umode_interp
+
+        Args:
+            keys : list of tuples
+                List of baseline-pol DataContainer tuples to operate on.
+            auto_keys : list of tuples
```

Comment:
I don't get this at all. For a given baseline, aren't there two auto-correlations to look at (one for each antenna)? And shouldn't that be the default behavior?